### PR TITLE
[ui] Query for automationPolicySensor on assets and asset groups

### PR DESCRIPTION
**Pull request by hellendag**
## Summary & Motivation

For assets and asset groups, query for `automationPolicySensor` to show the relevant sensor in the new AMP flagged state, instead of the global auto-materialization state tag.

Screenshot 2023-12-26 at 4 36 54 PM

## How I Tested These Changes

Added a Jest test for asset group tags.

View an asset group that has an AMP sensor. Verify that the tag appears.

View an asset that has an AMP sensor. Verify same.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #18900** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18900?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/18900?utm_source=stack-comment).

File: js_modules/dagster-ui/packages/ui-core/src/sensors/SensorRoot.tsx [hunk 1]
```diff
@@ -89,7 +89,10 @@ export const SensorRoot = ({repoAddress}: {repoAddress: RepoAddress}) => {
 
   if (!data && loading) {
     return (
-      <Box margin={{top: 32}} flex={{direction: 'row', alignItems: 'center', gap: 16}}>
+      <Box
+        margin={{top: 32}}
+        flex={{direction: 'row', alignItems: 'center', justifyContent: 'center', gap: 16}}
+      >
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/sensors/SensorRoot.tsx:95]: Unrelated fix to center the loading state.

File: js_modules/dagster-ui/packages/ui-core/src/assets/__fixtures__/AssetGroupTags.fixtures.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      selector: {
+        groupName: GROUP_NAME,
+        repositoryName: DUNDER_REPO_NAME,
+        repositoryLocationName: LOCATION_NAME,
+      },
+    },
+  },
+  result: {
+    data: {
+      __typename: 'Query',
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/__fixtures__/AssetGroupTags.fixtures.ts:59]: You should try `buildQueryMock` allows you to skip typing `__typename: Query` every time and slightly less boilerplate for mocking queries.

**Review by salazarm — APPROVED** [review]
